Source/WebCore/svg/SVGTransformList.cpp

```diff
@@ -62,54 +62,75 @@ AffineTransform SVGTransformList::concatenate() const
     return result;
 }
 
-template<typename CharacterType> bool SVGTransformList::parseGeneric(StringParsingBuffer<CharacterType>& buffer)
+template<typename CharacterType> bool SVGTransformList::parseGeneric(StringParsingBuffer<CharacterType>& buffer, ListReplacement listReplacement)
 {
     bool delimParsed = false;
     skipOptionalSVGSpaces(buffer);
 
-    while (buffer.hasCharactersRemaining()) {
-        delimParsed = false;
-        
-        auto parsedTransformType = SVGTransformable::parseTransformType(buffer);
-        if (!parsedTransformType)
-            return false;
+    auto parsingSucceeded = [&]() {
+        size_t itemIndex = 0;
+        auto currentListReplacement = listReplacement;
 
-        auto parsedTransformValue = SVGTransformable::parseTransformValue(*parsedTransformType, buffer);
-        if (!parsedTransformValue)
-            return false;
+        while (buffer.hasCharactersRemaining()) {
+            delimParsed = false;
 
-        append(SVGTransform::create(WTFMove(*parsedTransformValue)));
+            auto parsedTransformType = SVGTransformable::parseTransformType(buffer);
+            if (!parsedTransformType)
+                return false;
 
-        skipOptionalSVGSpaces(buffer);
+            RefPtr<SVGTransform> existingTransform;
+            if (currentListReplacement == ListReplacement::Replace && itemIndex < m_items.size() && parsedTransformType == m_items[itemIndex]->type())
+                existingTransform = m_items[itemIndex].ptr();
 
-        if (skipExactly(buffer, ','))
-            delimParsed = true;
+            RefPtr parsedTransform = SVGTransformable::parseTransform(*parsedTransformType, buffer, existingTransform);
+            if (!parsedTransform)
+                return false;
 
-        skipOptionalSVGSpaces(buffer);
-    }
-    return !delimParsed;
+            if (currentListReplacement == ListReplacement::Replace && parsedTransform.get() != existingTransform.get()) {
+                currentListReplacement = ListReplacement::Append;
+                resize(itemIndex);
+            }
+
+            if (currentListReplacement == ListReplacement::Append)
+                append(WTFMove(parsedTransform.releaseNonNull()));
+
+            skipOptionalSVGSpaces(buffer);
+
+            if (skipExactly(buffer, ','))
+                delimParsed = true;
+
+            skipOptionalSVGSpaces(buffer);
+
+            ++itemIndex;
+        }
+
+        if (currentListReplacement == ListReplacement::Replace && itemIndex < m_items.size())
+            resize(itemIndex);
+
+        return !delimParsed;
+    }();
+
+    if (!parsingSucceeded && listReplacement == ListReplacement::Replace)
+        clearItems();
```

Comment:
It is not clear why we `clearItems()` when an error happens but only when `ListReplacement::Replace`. I understand you want to keep the existing behavior. But I think moving `clearItems()` back to `SVGTransformList::parse(StringView value)` make reading this function easier.